src/Core/View.js

```diff
@@ -296,6 +296,13 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
         throw new Error(`${layer.protocol} is not a recognized protocol name.`);
     }
     layer = _preprocessLayer(this, layer, provider);
+    layer.whenReady.then(() => {
+        // if we still haven't any validExtent after getCapabilities, we assume
+        // it is the same as the configured extent
+        if (!layer.validExtent) {
+            layer.validExtent = layer.extent;
```

Comment:
and if `layer.extent` is undefined?

Reply:
Normally it shouldn't happen: if disableGetCap is true, we check for it. Otherwise the `parseExtent` method has several fallbacks that cannot reasonably fail.